util/converter/tarconverter/tarconverter.go

```diff
@@ -21,7 +22,13 @@ func NewReader(srcContent io.Reader, headerConverter HeaderConverter) io.ReadClo
 			if err == io.EOF {
 				// Signals end of archive.
 				rebasedTar.Close()
-				w.Close()
+				// drain the reader until hitting EOF
+				_, err = io.Copy(w, srcContent)
```

Comment:
The destination for the Copy should be to discard

Reply:
No, when something is appended after the tar ball, it has to be copied to the output

Reply:
I don't think that is correct. If there is unread padding then `w.Close()` will write the correct padding for the new file and padding from the old file should be ignored, not duplicated.

Reply:
I guess non-padding data can be potentially added?

cc @ktock

Reply:
Looks like same concept https://github.com/moby/moby/blob/v26.0.0/layer/layer_store.go#L251-L254

Reply:
Non-padding data should not cause `EOF` but an actual error about invalid data at the end of tar.

Reply:
fixed